### PR TITLE
Fix translation for Dock/Undock label in User-Defined dialog

### PR DESCRIPTION
**Pull request by ArkadiuszMichalski**
Fix [8020](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8020).

@donho 
I create additional function `getAttrNameByIdStr`. It's universal, we pass the `targetNode` for searching `<Item>` elements  (so before passing we can choose any one, regardless of the level of nesting). It retrieves the value of the `"name"` attribute based on the name and the value of the `"id"` attribute. Attribute names `"id"` and `"name"` can be changed (these are default).
I noticed that sometimes we have already translated value(s) in some node(s) and it doesn't make sense to add more nodes with the same value (especially when we are in the same parent). Unfortunately, there was no universal method defined to achieve this, so I added it.

### CODE REVIEW

File: PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp [hunk 1]
```diff
@@ -925,9 +925,14 @@ void UserDefineDialog::reloadLangCombo()
 void UserDefineDialog::changeStyle()
 {
     _status = !_status;
-    ::SetDlgItemText(_hSelf, IDC_DOCK_BUTTON, (_status == DOCK)?TEXT("Undock"):TEXT("Dock"));
-
-	auto style = ::GetWindowLongPtr(_hSelf, GWL_STYLE);
+    NativeLangSpeaker *pNativeSpeaker = (NppParameters::getInstance()).getNativeLangSpeaker();
+    TiXmlNodeA *targetNode = nullptr;
+    pNativeSpeaker->getNativeLangA() && (targetNode = (pNativeSpeaker->getNativeLangA())->FirstChildElement("Dialog")) && (targetNode = targetNode->FirstChildElement("UserDefine"));
```
Comment (chcg) [inline PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp:930]: C:\projects\notepad-plus-plus\PowerEditor\src\ScintillaComponent\UserDefineDialog.cpp(930): error C2220: the following warning is treated as an error [C:\projects\notepad-plus-plus\PowerEditor\visual.net\notepadPlus.vcxproj]
     5>C:\projects\notepad-plus-plus\PowerEditor\src\ScintillaComponent\UserDefineDialog.cpp(930): warning C4706: assignment within conditional expression [C:\projects\notepad-plus-plus\PowerEditor\visual.net\notepadPlus.vcxproj]

Comment (ArkadiuszMichalski) [reply PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp]: @chcg 
Is there any equivalent in GCC for this warning ([manual](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#Warning-Options))? For `Debug` we turn on a lot of things but I don't see any such message: https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/46232207/job/8aqqrg6cde100t7m.

File: PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp [hunk 2]
```diff
@@ -925,9 +925,19 @@ void UserDefineDialog::reloadLangCombo()
...
+    TiXmlNodeA *targetNode = nullptr;
+    if (pNativeSpeaker->getNativeLangA())
+    {
+        targetNode = (pNativeSpeaker->getNativeLangA())->FirstChildElement("Dialog");
+        if (targetNode)
+            targetNode = targetNode->FirstChildElement("UserDefine");
+    }
+    generic_string dockStr = pNativeSpeaker->getAttrNameByIdStr(TEXT("Dock"), targetNode, std::to_string(IDC_DOCK_BUTTON).c_str());
+    generic_string undockStr = pNativeSpeaker->getAttrNameByIdStr(TEXT("Undock"), targetNode, std::to_string(IDC_UNDOCK_BUTTON).c_str());
+    ::SetDlgItemText(_hSelf, IDC_DOCK_BUTTON, (_status == DOCK) ? undockStr.c_str() : dockStr.c_str());
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp:938]: I would do somethings similar as following code:
```cpp
generic_string dockButtonLabel;
generic_string defauleLabel;
generic_string idStr;
if (_status == DOCK)
{
    defauleLabel = TEXT("Undock");
    idStr = std::to_string(IDC_UNDOCK_BUTTON);
}
else
{
    defauleLabel = TEXT("Dock");
    idStr = std::to_string(IDC_DOCK_BUTTON);
}

dockButtonLabel= pNativeSpeaker->getAttrNameByIdStr(defauleLabel.c_str(), targetNode, idStr.c_str());
::SetDlgItemText(_hSelf, IDC_DOCK_BUTTON, dockButtonLabel.c_str());
```

So we don't call `getAttrNameByIdStr` twice.

**Review by donho — CHANGES_REQUESTED** [review]
Please follow the suggestion.

**Review by donho — APPROVED** [review]

**Pull request closed without merging**